@here/harp-mapview/lib/poi/PoiRenderer.ts

```diff
@@ -546,14 +538,21 @@ export class PoiRenderer {
         }
 
         if (opacity > 0) {
-            const layer = this.addLayer(poiInfo.renderOrder!);
-            this.m_poiBatchRegistry.addPoi(
-                poiInfo,
-                this.m_tempScreenBox,
-                viewDistance,
-                opacity,
-                layer
-            );
+            if (!poiInfo.buffer) {
+                this.preparePoi(poiInfo.textElement, env);
+            } else if (
+                poiInfo.buffer &&
+                this.m_layers.findIndex(layer => {
+                    return layer === poiInfo.buffer?.layer;
+                }) < 0
```

Comment:
Now that we don't destroy the PoiRenderer, I don't see when can poiInfo end up with a layer that doesn't exist anymore. I don't see that we ever delete a layer.

Reply:
right, this was the first approach I added to solve the issue, I ll remove it, now that its not needed anymore